privval/secret_connection.go

```diff
@@ -339,14 +344,14 @@ func shareEphPubKey(conn io.ReadWriter, locEphPub *[32]byte) (remEphPub *[32]byt
 func deriveSecrets(
 	dhSecret *[32]byte,
 	locIsLeast bool,
-) (recvSecret, sendSecret *[aeadKeySize]byte) {
+) (recvSecret, sendSecret *[aeadKeySize]byte, err error) {
```

Comment:
perhaps defining a 'secretPair' type would make it easier to handle the multiple returns from this function.

Reply:
Agreed. Though I wouldn't object to doing that separately.

(The fact that we're handling pointers to arrays here is also kind of janky, and a wrapper type would let us avoid that)

Reply:
I think this (given that its not exported, and internal) isn't a priority for this review.